### PR TITLE
Changes made to connectivity/check/logging.go

### PR DESCRIPTION
**Pull request by AHB102**
Hi there @joestringer, I forgot to add a PR, resolved issue #2042, added context as a parameter to the failCommon method, to allow each test caller to pass their own context down when invoking Fail(), Failf(), Fatal(), and Fatalf()

### CODE REVIEW

**Comment by joestringer** [comment]
Thanks for the PR. It doesn't build, could you fix, build and test it and re-push a fresh version?

**Comment by AHB102** [comment]
I have some questions which might not be fully within the scope of this issue, so can I message you on slack ?

**Comment by joestringer** [comment]
I would suggest posting in #development channel on Slack. I am not deeply involved with cilium-cli, I just noticed the issue. Posting in that public channel should provide opportunities for other developers in the community to see your messages and provide input.

File: connectivity/check/logging.go [hunk 1]
```diff
@@ -285,7 +285,7 @@ func (t *Test) Infof(format string, a ...interface{}) {
 	t.logf(info+" "+format, a...)
 }
 
-func (t *Test) failCommon() {
+func (t *Test) failCommon(ctx context.Context) {
```
Comment (christarazi) [inline connectivity/check/logging.go:288]: This `ctx` is not used after being passed in. It seems like the `context.Background()` needs to be replaced with it on line 298.

**Review by christarazi — CHANGES_REQUESTED** [review]
Thanks for the PR. A comment below to address.

Additionally, as mentioned in the issue, please link the issue in the description of the commit msg. You can also squash the commits into one as there is no need for multiple commits here. For more info on how to format commit msgs, please see https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submit-pr.

**Comment by AHB102** [comment]
@christarazi Thanks for the comment! I'm quite new to open source, but I'm slowly but surely getting the hang of it.

File: connectivity/check/logging.go [hunk 2]
```diff
@@ -312,7 +312,8 @@ func (t *Test) failCommon() {
 // will go directly to the user-specified writer.
 func (t *Test) Fail(a ...interface{}) {
 	t.log(fail, a...)
-	t.failCommon()
+	t.failCommon(t.ctx.Context())
+
```
Comment (christarazi) [inline connectivity/check/logging.go:316]: Nit: extra line here that's not necessary

**Review by christarazi — COMMENTED** [review]
Thanks!

Could you reword the title of the commit msg to be:

```
connectivity: Plumb context for SIGINT notifications
```

**Comment by AHB102** [comment]
@christarazi  I have reworded the commit message and removed the extra line at line 316.

**Review by christarazi — APPROVED** [review]

**Comment by christarazi** [comment]
@AHB102 Seems like the code has some compilation errors.

**Pull request closed without merging**